### ISSUE TITLE
BUG: Scripts with dll dependencies are uncachable.

### ISSUE THREAD

**Issue by pms1969** [fsharp/FAKE#1113]
I have a build.fsx with a number of dll dependencies.  Some home grown things as well as SharpSvn.dll

when I run a simple target from this script, say something that just prints out "hello world!"  I would expect that after the first run, that my script is compiled and cached in the .fake folder.

However when I run my script a second time, I get an error stating that it is unable to access the cached dll.

some of the sanitised output here 
```
FakePath: C:\UBS\Dev\Delta\git\FAKE\src\app\FAKE\bin\Debug\FakeLib.dll
FAKE - F# Make "4.17.0"
Running Buildscript: build.fsx
Using cache

CACHING WARNING
this might happen after Updates...
please open a issue on FAKE and /cc @matthid ONLY IF this happens reproducibly)

Error: System.Reflection.ReflectionTypeLoadException: Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information.
   at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
   at System.Reflection.RuntimeModule.GetTypes()
   at System.Reflection.Assembly.GetTypes()
   at Fake.FSIHelper.runScriptCached@309.Invoke(Unit unitVar0) in C:\......\FAKE\src\app\FakeLib\FSIHelper.fs:line 311

   at Yaaf.FSharp.Scripting.Helper.consoleCapture[a](TextWriter out, TextWriter err, FSharpFunc`2 f) in C:\......\FAKE\paket-files\matthid\Yaaf.FSharp.Scripting\src\source\Yaaf.FSharp.Scripting\YaafFSharpScripting.fs:line 1018
   at Fake.FSIHelper.runScriptCached(Boolean printDetails, CacheInfo cacheInfo, TextWriter out, TextWriter err) in C:\......\FAKE\src\app\FakeLib\FSIHelper.fs:line 309
   at Fake.FSIHelper.runFAKEScriptWithFsiArgsAndRedirectMessages(Boolean printDetails, FsiArgs _arg1, IEnumerable`1 env, FSharpFunc`2 onErrMsg, FSharpFunc`2 onOutMsg, Boolean useCache) in C:\......\FAKE\src\app\FakeLib\FSIHelper.fs:line 495
Cache is invalid, recompiling
Unable to access C:\......\mysolution\trunk\.fake\build.fsx_C1BEEB8D.dll
Building project with version:
```

I've tracked down the issue to an issue loading the assembly in FSIHelper.fs `runScriptCached` function.  When it tries to get the types from the created assembly, it barfs.  see: http://stackoverflow.com/questions/7889228/how-to-prevent-reflectiontypeloadexception-when-calling-assembly-gettypes 

I've fiddled with the code a bit, and believe it to be because it is unable to load our other assemblies, because they are not in the immediate path...

**Comment by pms1969** [comment]
I hope that makes a modicum of sense.

**Comment by forki** [comment]
I like that "/cc @matthid" part ;-)

**Comment by pms1969** [comment]
indeed.  sorry, I've been looking at this for so long I grew immune to it.

**Comment by matthid** [comment]
> because it is unable to load our other assemblies, because they are not in the immediate path...

that shouldn't matter because we load all assemblies previosly and install an assembly resolve handler... I think we should add the code you linked on stackoverflow anyway... is there a way to reproduce this?

**Comment by pms1969** [comment]
Fully agree that that stackoverflow code should be added, but it doesn't stop this problem from occurring (tried it).

It works for the case when it's creating the cached dll, just not when loading it back.  I can confirm that it's the lack of the DLL's, since adding them to the .fake directory allows the cached dll to then load and execute as expected.

When I get to work, I'll try and put together a little script that replicates the issue

**Comment by pms1969** [comment]
well, I stand corrected.  for the life of me, I can't reproduce this today.  I'll just crawl back into my hole :-s

**Comment by matthid** [comment]
Well reproducable means the error happens every second time (because we cear the cache on errors the next run will be successful). Did the error disappear on your original `build.fsx` as well?

**Comment by pms1969** [comment]
sorry, missed this....  It did disappear on the original as well.  I tried getting a small cut down version of the script that represented what was happening, but just couldn't get it to do what it was doing...  Switched back to the build.fsx that this was happening on when I raised the issue, and it was working as expected.  No idea what happened.  Some workstation state obviously in a twist.  If I come across it again, I'll be sure to let you know.

BTW, it was happening every single time, not every second time.  What I observed was Fake trying to load the cached version, failing, caching the new (but unchanged) version of the script, and continuing execution.  Next time it ran, exactly the same thing.
